agent/api/task/task.go

```diff
@@ -2548,3 +2556,62 @@ func getDomainForPartition(region string) string {
 	}
 	return partition.DNSSuffix()
 }
+
+func (task *Task) requireEnvfiles() bool {
+	for _, container := range task.Containers {
+		if container.ShouldCreateWithEnvFiles() {
+			return true
+		}
+	}
+	return false
+}
+
+func (task *Task) initializeEnvfilesResource(config *config.Config, credentialsManager credentials.Manager, ) error {
+
+	for _, container := range task.Containers {
+		envfileResource, err := envFiles.NewEnvironmentFileResource(config.Cluster, task.Arn, config.AWSRegion, config.DataDir,
+			container.EnvironmentFiles, credentialsManager, task.ExecutionCredentialsID)
+		if err != nil {
+			return errors.Wrap(err, "unable to initialize envfiles resource")
+		}
+		task.AddResource(envFiles.ResourceName, envfileResource)
+		container.BuildResourceDependency(envfileResource.GetName(), resourcestatus.ResourceCreated, apicontainerstatus.ContainerCreated)
+	}
+
+	return nil
+}
+
+func (task *Task) getEnvfilesResource() ([]taskresource.TaskResource, bool) {
+	task.lock.RLock()
+	defer task.lock.RUnlock()
+
+	res, ok := task.ResourcesMapUnsafe[envFiles.ResourceName]
+	return res, ok
+}
+
+// ReadEnvVarsFromEnvfiles should be called when creating a container -
+// this method reads the environment variables specified in the environment files
+// that was downloaded to disk and merges it with existing environment variables
+func (task *Task) ReadEnvVarsFromEnvfiles(container *apicontainer.Container) *apierrors.ResourceInitError {
+	var envfileResource *envFiles.EnvironmentFileResource
+	if container.ShouldCreateWithEnvFiles() {
+		resource, ok := task.getEnvfilesResource()
+		if !ok {
+			err := errors.New("task environment files: unable to retrieve environment files resource")
+			return apierrors.NewResourceInitError(task.Arn, err)
+		}
+		envfileResource = resource[0].(*envFiles.EnvironmentFileResource)
```

Comment:
in initializeEnvfilesResource the envfileResource is added per container, but here you only get the first one, which doesn't seem right?

Reply:
Also based on the information in the env file task resource and how you add task resource to the **ResourcesMapUnsafe**, don't seem there is an easy way for you to figure out which task resource maps to which container.

Reply:
yep realize i am missing some logic to make sure envfiles only get sent to corresponding containers, will update.